src/main/java/org/aksw/simba/lemming/ColouredGraphDecorator.java

```diff
@@ -0,0 +1,300 @@
+     * 
+     * @param tailId
+     * @param headId
+     * @param edgeColour
+     * @return int - the edgeId of newly added edge
+     */
+    @Override
+    public int addEdge(int tailId, int headId, BitSet edgeColour) {
+        return this.graph.addEdge(tailId, headId, edgeColour);
+    }
+
+    /**
+     * Remove edge from the graph
+     * 
+     * @param edgeId
+     */
+    @Override
+    public void removeEdge(int edgeId) {
+        this.graph.removeEdge(edgeId);
+    }
+
+    /**
+     * Get Set of all edges in the graph
+     * 
+     * @return IntSet
+     */
+    @Override
+    public IntSet getEdges() {
+        return this.graph.getEdges();
+    }
+
+    /**
+     * Get edge color
+     * 
+     * @param edgeId
+     * @return BitSet
+     */
+    @Override
+    public BitSet getEdgeColour(int edgeId) {
+        return this.graph.getEdgeColour(edgeId);
+    }
+
+    /**
+     * Get property color
+     * 
+     * @return Object
+     */
+    @Override
+    public Object getRDFTypePropertyColour() {
+        return this.graph.getRDFTypePropertyColour();
+    }
+
+    /**
+     * Get the vertex id of the tail if the edge
+     * 
+     * @param edgeId
+     * @return int
+     */
+    @Override
+    public int getTailOfTheEdge(int edgeId) {
+        return this.graph.getTailOfTheEdge(edgeId);
+    }
+
+    /**
+     * Get the vertex id of the head if the edge
+     * 
+     * @param edgeId
+     * @return int
+     */
+    @Override
+    public int getHeadOfTheEdge(int edgeId) {
+        return this.graph.getHeadOfTheEdge(edgeId);
+    }
+
+    /**
+     * Get list of all vertex IDs connecting to the the edgeId
+     *
+     * @param edgeId - the id of an edge connecting the vertices together
+     * @return set of vertex ID's
+     */
+    @Override
+    public IntSet getVerticesIncidentToEdge(int edgeId) {
+        return this.graph.getVerticesIncidentToEdge(edgeId);
+    }
+
+    /**
+     * Get set of all in neighbors to a vertex
+     * 
+     * @param vertexId
+     * @return set of all in neighbors
+     */
+    @Override
+    public IntSet getInNeighbors(int vertexId) {
+        return this.graph.getInNeighbors(vertexId);
+    }
+
+    /**
+     * Get set of all out neighbors to a vertex
+     * 
+     * @param vertexId
+     * @return set of all out neighbors
+     */
+    @Override
+    public IntSet getOutNeighbors(int vertexId) {
+        return this.graph.getOutNeighbors(vertexId);
+    }
+
+    /**
+     * Get set of all vertices
+     * 
+     * @return set of all vertices
+     */
+    @Override
+    public IntSet getVertices() {
+        return this.graph.getVertices();
+    }
+
+    /**
+     * Get number of edges between two vertices
+     * 
+     * @return int - number of edges
+     */
+    public int getNumberOfEdgesBetweenVertices() {
+        return IntSetUtil.intersection(this.graph.getEdgesIncidentTo(this.triple.tailId),
+                this.graph.getEdgesIncidentTo(this.triple.headId)).size();
+    }
```

Comment:
1. From the javadoc comment, I do not get the meaning of this method. Is it the number of edges between any two vertices or between two specific vertices?
2. The implementation is not really efficient. It creates a new IntSet just to recognize the size of it. After that, it will throw the set away. That can be done much better :wink:

Reply:
Improved the implementaion and the javadoc.